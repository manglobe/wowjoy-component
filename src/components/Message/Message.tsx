import * as React from "react";
import styled from "styled-components";
const Wrap = styled.div<{ defaultStyles?: string }>`
  width: 480px;
  height: 36px;
  line-height: 36px;
  border: 1px solid transparent;
  font-size: 12px;
  ${(props: { defaultStyles?: string }) => props.defaultStyles};
`;
export interface Props {
  className?: string;
  defaultStyles?: string;
  icon?: React.ReactNode;
  children?: React.ReactNode;
  onClose?: (
    e: React.MouseEvent<HTMLSpanElement, MouseEvent>
  ) => boolean | void;
}
class Message extends React.PureComponent<Props, {}> {
  closeHandle = (e: React.MouseEvent<HTMLSpanElement, MouseEvent>) => {
    const { onClose } = this.props;
    if (onClose && onClose(e) === false) {
      return;
    }
  };
  render() {
    const { className, defaultStyles, icon, children, onClose } = this.props;
    return (
      <Wrap
        defaultStyles={defaultStyles}
        className={"wjc-message " + className || ""}
      >
        {icon}
        {children}
        {onClose && (
          <i onClick={this.closeHandle} className={"wjc-message-close"} />
        )}
      </Wrap>
    );
  }
}

export default Message;
