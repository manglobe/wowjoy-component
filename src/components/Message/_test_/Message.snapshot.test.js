import React from "react";
import Message from "../index.js";
import renderer from "react-test-renderer";
import * as TestUtils from "react-dom/test-utils";
import { MemoryRouter, Route } from "react-router-dom";
import { mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

const { renderIntoDocument } = TestUtils;

describe("Message", () => {
  const reactObj = <Message>Message Content</Message>;
  it("renders correctly", () => {
    const tree = renderer.create(reactObj).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("render icon correctly", () => {
    const reactObj = <Message icon={<span className="icon">icon</span>} />;
    const tree = renderer.create(reactObj).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("render i.wjc-message-close correctly", () => {
    const reactObj = <Message onClose={() => {}} />;
    const tree = renderer.create(reactObj).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
