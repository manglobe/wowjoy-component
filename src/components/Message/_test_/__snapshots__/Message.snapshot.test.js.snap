// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message render i.wjc-message-close correctly 1`] = `
<div
  className="wjc-message undefined sc-bdVaJa EKEAs"
>
  <i
    className="wjc-message-close"
    onClick={[Function]}
  />
</div>
`;

exports[`Message render icon correctly 1`] = `
<div
  className="wjc-message undefined sc-bdVaJa EKEAs"
>
  <span
    className="icon"
  >
    icon
  </span>
</div>
`;

exports[`Message renders correctly 1`] = `
<div
  className="wjc-message undefined sc-bdVaJa EKEAs"
>
  Message Content
</div>
`;
